dataactbroker/handlers/fileHandler.py

```diff
@@ -890,6 +888,131 @@ def generateFile(self):
+                      job_status_id=JOB_STATUS_DICT['running'], job_type_id=JOB_TYPE_DICT['file_upload'],
+                      user_id=user_id, file_type_id=FILE_TYPE_DICT_LETTER_ID[file_type], start_date=start_date,
+                      end_date=end_date)
+
+        # need to add & commit the new job first so we can reference the job_id below for the task generation
+        sess.add(new_job)
+        sess.commit()
+
+        task_key = uuid4()
+        task = FileGenerationTask(generation_task_key=task_key, job_id=new_job.job_id)
+        sess.add(task)
+        sess.commit()
+
+        if not self.isLocal:
+            # Create file D API URL with dates and callback URL
+            get_url = FileHandler.get_d_file_url(task_key, file_type_name, cgac_code, start_date, end_date)
+
+            _debug_logger.debug('Calling Detached D file API => %s', get_url)
+            try:
+                if not self.call_d_file_api(get_url):
+                    # If the call to the external API wasn't successful, mark the job as failed
+                    new_job.error_message = "%s data unavailable for the specified date range" % file_type
+                    new_job.job_status_id = JOB_STATUS_DICT['failed']
+                    sess.commit()
+            except Timeout as e:
+                exc = ResponseException(str(e), StatusCode.CLIENT_ERROR, Timeout)
+                return False, JsonResponse.error(e, exc.status, url="", start="", end="", file_type=file_type)
+        else:
+            self.complete_generation(task.generation_task_key, file_type)
+
+        # Return same response as check generation route
+        return self.check_detached_generation(user_id, file_type)
+
+    def check_detached_generation(self, user_id=None, file_type=None):
+        """ Return information about file generation jobs
+
+        Returns:
+            Response object with keys status, file_type, url, message.
+            If file_type is D1 or D2, also includes start and end.
+        """
+        if user_id is None:
+            user_id = LoginSession.getName(session)
+
+        if file_type is None:
+            requestDict = RequestDictionary(self.request)
+            if not requestDict.exists("file_type"):
+                raise ResponseException("Check detached generation route requires file_type",
+                                        StatusCode.CLIENT_ERROR)
+
+            file_type = requestDict.getValue("file_type")
+
+        sess = GlobalDB.db().session
+
+        # Get the last job created by the specified user (only care about the jobs that have a user id here
+        # since the generation is detached from a submission)
+        uploadJob = sess.query(Job).filter_by(
+            user_id=user_id,
+            file_type_id=FILE_TYPE_DICT_LETTER_ID[file_type],
+            job_type_id=JOB_TYPE_DICT['file_upload']
+        ).order_by(Job.created_at.desc()).first()
```

Comment:
This means that a user can't retrieve multiple submissions, right? Would it make sense to keep track of the job_id instead? Then a user could have multiple jobs per file type at once.